### ISSUE TITLE
Stop Automatic Scroll on Cell Update?

### ISSUE THREAD

**Issue by cschraga** [patchthecode/JTAppleCalendar#637]
Im using version 7.1.1

I'm trying to create an iPhone drag-and-drop scheduling application using your calendar. Idea being you pick up a cell from a uitable view and drag it onto your JTAppleCalendar.  Apple doesn't yet have drag-and-drop functionality for iPhone so I recreated it.  Anyway, everything is working smoothly EXCEPT every time I run reloadItems() on the JTAppleCalendarView it automatically scrolls to the top.

Any advice on how I can maintain the current contentOffset of the JTAppleCalendarView?  It'll make it so much nicer.

Anyway, heres some screen shots just so you can get a sense of what I'm trying to do:

**Here's the pre-drag screen shot:**
Everything is a-ok here

![jtss1](https://user-images.githubusercontent.com/15016969/33282664-52170580-d377-11e7-976f-067175fba4e3.jpg)

**Mid drag:**
Everything still working fine

![jtss2](https://user-images.githubusercontent.com/15016969/33282698-6c79f824-d377-11e7-86e0-a40dc6ff40df.jpg)

**Post drag:**
**_Here's where the problem lies_**
Once the drag is complete, I run reloadItems() on the date that the user dropped the cell in. When reloadItems() loads, the CalendarView scrolled up to October which is the start date in my configuration parameters. This is what I'm trying to fix.

![image](https://user-images.githubusercontent.com/15016969/33282736-8e010258-d377-11e7-85b2-60d012057322.png)

**Date zoom:**

This has nothing to do with the problem I just wanted to show you because i think its cool.  You can click on a date and it zooms in. Since its hard to see what's scheduled while looking at the whole calendar by month:

![jtss3](https://user-images.githubusercontent.com/15016969/33282809-ccb73698-d377-11e7-8c38-a2a40ecbbc3b.jpg)

Make sense?
Any help would be greatly appreciated.

**Comment by patchthecode** [comment]
Good job man. Looks good.

when ever you reload the calendar, if you have made any changes which affects the layout of the cells, it will reset the calendar, because your cells will have to be recreated.

Things that causes calendar refresh are if any of the following changes:
1. StartDate or EndDate
2. minimumInteritemSpacing
3. minimumLineSpacing
4. sectionInset
5. sizeOfMonth
6. allowsDateCellStretching
7. scrollDirection
8. width/height of calendar
9. numberOfRows
10. generateInDates/generateOutDates
11. hasStrictBoundaries
etc etc

If you want to reload without scrolling, you can try something like
```swift
reloadData(withAnchor: dateToAnchorTo)
```
let me know if this works for you?

**Comment by cschraga** [comment]
Worked beautifully. Thanks!

**Comment by cschraga** [comment]
Sorry, I know this is closed but one more quick question... is there a way of resizing the calendar view or its superview without it reloading? I have my calendar in a resizable view and want to keep it static while resizing.

**Comment by patchthecode** [comment]
you want to resize the calendarView... ok.
WHen you resize the calendar view, should the cells shrink to match the view?

**Comment by patchthecode** [comment]
Also, open a new issue for this thanks.
